docs/streams/developer-guide.html

```diff
@@ -267,21 +295,532 @@
+
+            <pre class="brush: java;">
+            KStream&lt;String, Long&gt; stream = ...;
+            KStream&lt;String, Long&gt;[] branches = stream.branch(
+		            (key, value) -> key.startsWith("A"), /* first predicate  */
+		            (key, value) -> key.startsWith("B"), /* second predicate */
+		            (key, value) -> true                 /* third predicate  */
+            );
+            // KStream branches[0] contains all records whose keys start with "A"
+            // KStream branches[1] contains all records whose keys start with "B"
+            // KStream branches[2] contains all other records
+            // Java 7 example: cf. `filter` for how to create `Predicate` instances
+	        </pre>
+        </td>
+    </tr>
+	 <tr>
+	   <td><b>Filter</b>: <code>KStream->KStream or KTable->KTable</code></td>
+	   <td>
+	     Evaluates a boolean function for each element and retains those for which the function returns true. 
+
+	     <pre class="brush: java;">
+             KStream&lt;String, Long&gt; stream = ...;
+
+             // A filter that selects (keeps) only positive numbers
+             // Java 8+ example, using lambda expressions
+             KStream&lt;String, Long&gt; onlyPositives = stream.filter((key, value) -> value > 0);
+
+             // Java 7 example
+             KStream&lt;String, Long&gt; onlyPositives = stream.filter(
+               new Predicate&lt;String, Long&gt;() {
+                 @Override
+                 public boolean test(String key, Long value) {
+                   return value > 0;
+                 }
+               });
+	     </pre>
+	   </td>
+	 </tr>
+	 <tr>
+	   <td><b>Inverse Filter</b>: <code>KStream->KStream or KTable->KTable</code></td>
+	   <td>
+	     Evaluates a boolean function for each element and drops those for which the function returns true.
+
+	     <pre class="brush: java;">
+             KStream&lt;String, Long&gt; stream = ...;
+		 
+             // An inverse filter that discards any negative numbers or zero
+             // Java 8+ example, using lambda expressions
+             KStream&lt;String, Long&gt; onlyPositives = stream.filterNot((key, value) -> value <= 0);
+
+             // Java 7 example
+             KStream&lt;String, Long&gt; onlyPositives = stream.filterNot(
+              new Predicate&lt;String, Long&gt;() {
+                @Override
+                public boolean test(String key, Long value) {
+                    return value <= 0;
+                }
+             });
+         </pre>
+	      </td>
+	 </tr>
+	 <tr>
+	   <td><b>FlatMap</b>: <code>KStream->KStream </code></td>
+         <td>
+             Takes one record and produces zero, one, or more records. You can modify the record keys and values,
+             including their types.<p>
+
+             Marks the stream for data re-partitioning: Applying a grouping or a join after <code>flatMap</code> will result in
```

Comment:
I'd definitely highlight "Marks the stream for data re-partitioning:" by making it bold font!  Same for the other instances of "Marks the stream" throughout this doc.